ddtrace/contrib/cherrypy/middleware.py

```diff
@@ -68,11 +68,12 @@ def _setup(self):
         cherrypy.request.hooks.attach("after_error_response", self._after_error_response, priority=5)
 
     def _on_start_resource(self):
-        if self.use_distributed_tracing:
-            context = HTTPPropagator.extract(cherrypy.request.headers)
-            # Only need to activate the new context if something was propagated
-            if context.trace_id:
-                self._tracer.context_provider.activate(context)
+        trace_utils.activate_distributed_headers(
+            self._tracer,
+            int_config=config.cherrypy,
+            request_headers=cherrypy.request.headers,
+            override=self.use_distributed_tracing,
```

Comment:
I'm a bit concerned about making this change. It is functionally equivalent but a bit misleading because `self.use_distributed_tracing` is always going to be non-None and hence its value is always going to be used (int_config is always going to be ignored).

@Yun-Kim WDYT about making all the FIXME changes before this change?

Reply:
Seems like I've inadvertently split these changes up into each separate integration (#2235, #2236, #2287, #2291, #2292, #2293, #2294, #2299) including the addition of `int_config` and the `distributed_tracing_enabled` env variable as well as the refactoring for `activate_distributed_headers()`, so we can close this PR now 😄